### ISSUE TITLE
Crash on breakpoint if freed node exists

### ISSUE THREAD

**Issue by miv391** [godotengine/godot#80442]
### Godot version

v4.2.dev2.official [da81ca62a]

### System information

Godot v4.2.dev2 - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 6GB (NVIDIA; 31.0.15.3640) - Intel(R) Core(TM) i5-3570K CPU @ 3.40GHz (4 Threads)

### Issue description

If a freed node exists when editor breaks on breakpoint, Godot crashes. My guess is that debug view tries to show the freed node and somehow fails. If the freed node is set to null before the breakpoint, crash doesn't happen.

`v4.1.1.stable.official` works fine, but `v4.2.dev2.official` crashes.

### Steps to reproduce

```gdscript
extends Node2D

func _ready() -> void:
	var node = Node.new()
	node.free()

	#node = null  # uncomment this line to prevent crash

	pass # add breakpoint here to crash
```

### Minimal reproduction project

[g42_crash.zip](https://github.com/godotengine/godot/files/12300368/g42_crash.zip)

**Comment by akien-mga** [comment]
Confirmed on Linux with f7bc653cbe81018fe362472a0143b7153a52f929:
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.2.dev.custom_build (f7bc653cbe81018fe362472a0143b7153a52f929)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib64/libc.so.6(+0x36980) [0x7efc8ab9b980] (??:0)
[2] EncodedObjectAsID::get_object_id() const (/home/akien/Projects/godot/godot.git/./core/io/marshalls.cpp:52)
[3] EditorDebuggerInspector::add_stack_variable(Array const&) (/home/akien/Projects/godot/godot.git/./editor/debugger/editor_debugger_inspector.cpp:236 (discriminator 2))
[4] ScriptEditorDebugger::_parse_message(String const&, unsigned long, Array const&) (/home/akien/Projects/godot/godot.git/./editor/debugger/script_editor_debugger.cpp:466)
[5] ScriptEditorDebugger::_notification(int) (/home/akien/Projects/godot/godot.git/./editor/debugger/script_editor_debugger.cpp:939 (discriminator 14))
[6] ScriptEditorDebugger::_notificationv(int, bool) (/home/akien/Projects/godot/godot.git/./editor/debugger/script_editor_debugger.h:60 (discriminator 14))
[7] Object::notification(int, bool) (/home/akien/Projects/godot/godot.git/./core/object/object.cpp:800)
[8] SceneTree::_process_group(SceneTree::ProcessGroup*, bool) (/home/akien/Projects/godot/godot.git/./scene/main/scene_tree.cpp:948)
[9] SceneTree::_process(bool) (/home/akien/Projects/godot/godot.git/./scene/main/scene_tree.cpp:1020 (discriminator 2))
[10] SceneTree::process(double) (/home/akien/Projects/godot/godot.git/./scene/main/scene_tree.cpp:510)
[11] Main::iteration() (/home/akien/Projects/godot/godot.git/main/main.cpp:3452)
[12] OS_LinuxBSD::run() (/home/akien/Projects/godot/godot.git/platform/linuxbsd/os_linuxbsd.cpp:912)
[13] godot-git(main+0x15a) [0x5761e00] (/home/akien/Projects/godot/godot.git/platform/linuxbsd/godot_linuxbsd.cpp:76)
[14] /lib64/libc.so.6(+0x236b7) [0x7efc8ab886b7] (??:0)
[15] /lib64/libc.so.6(__libc_start_main+0x85) [0x7efc8ab88775] (??:0)
[16] godot-git(_start+0x21) [0x5761be1] (??:?)
-- END OF BACKTRACE --
================================================================
Aborted (core dumped
```

Probably the same bug as #80236.

**Comment by akien-mga** [comment]
Fixed by #80256.
